### DISCUSSION TITLE
Moving edge

### DISCUSSION THREAD

**Discussion by NourEldin-Ali** [imixs/open-bpmn#165] [Q&A] [answered]
Hi @rsoika,

I tried to move a task that contains some elements, but when I change the position the edge are not moved correctly.

https://user-images.githubusercontent.com/61513661/215852352-1f848ee8-6cf6-4d2f-af04-8b785dc918dc.mp4

I'm trying to do like you did in move the lane in all the component.

https://user-images.githubusercontent.com/61513661/215852526-4345d627-33a6-40b6-96ec-0a66f7cc5123.mp4

Could you please give me an idea how you did it? how I can move the task with the all the componenet?
(I modified [BPMNChangeBoundsOperationHandler.java](https://github.com/imixs/open-bpmn/blob/master/open-bpmn.glsp-server/src/main/java/org/openbpmn/glsp/operations/BPMNChangeBoundsOperationHandler.java) that can allows me to move the elementNode in the task, but it's not clear to how i can move the elementEdge)

Thank your in advance for your reply

**Comment by rsoika** [comment, marked as answer]
Hi @NourEldin-Ali,
it looks to me that your edge is not part of the container element but part of the root node list in your model.
So when you add your gateway and task element as a 'child' element to the outer task-container, make sure that also your edge is part of this list.

**Reply by NourEldin-Ali** [reply]
Thank you

**Reply by rsoika** [reply]
@NourEldin-Ali, it looks like you try to implement the Sub-Process feature? I think it would be great to have this feature first in the open-bpmn.metamodel project? What did you think?

**Reply by NourEldin-Ali** [reply]
It's something like Sub-Process, but it's not.
I'm working in my PhD to do an extension of BPMN, but I think before go to the sub-process (If you need to expand and collapse the task) it's important to add ELK to allows the auto-layouting to easy use the tool.
